cmd/zfs/zfs_main.c

```diff
@@ -509,6 +512,28 @@ parseprop(nvlist_t *props, char *propname)
 	return (0);
 }
 
+/*
+ * Take a property name argument and add it to the given nvlist.
+ * Modifies the argument inplace.
+ */
+static int
+parsepropname(nvlist_t *props, char *propname)
```

Comment:
seems like this could return B_TRUE or B_FALSE, aka a boolean_t as it's a failed or not type of return code.
This will also allow you to do "if (parsepropname(p, p2))" - without checking the return value. Looks like you were trying to be consistent with parseprop() which arguably should have been doing the same thing